File: src/test/java/de/weimarnetz/registrator/controller/RegistratorControllerIT.java
package de.weimarnetz.registrator.controller;


import static org.assertj.core.api.Assertions.assertThat;

import java.util.Map;

import javax.inject.Inject;

import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.test.context.junit.jupiter.SpringExtension;

import de.weimarnetz.registrator.RegistratorApplication;

@ExtendWith(SpringExtension.class)
@SpringBootTest(classes = RegistratorApplication.class)
class RegistratorControllerIT {

    @Inject
    private RegistratorController registratorController;

    @Test
    void getTime() {
        // when
        ResponseEntity<Map<String, Long>> time = registratorController.getTime();

        // then
        assertThat(time.getStatusCode()).isEqualByComparingTo(HttpStatus.OK);
    }

}

File: src/test/java/de/weimarnetz/registrator/controller/RegistratorControllerIT.kt
package de.weimarnetz.registrator.controller

import de.weimarnetz.registrator.IntegrationTestConfig
import org.assertj.core.api.Assertions
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.HttpStatus

internal class RegistratorControllerIT(
    @Autowired
    private val registratorController: RegistratorController
) : IntegrationTestConfig() {
    @Test
    fun getTime() {
        // when
        val time = registratorController.getTime()

        // then
        Assertions.assertThat(time.statusCode).isEqualTo(HttpStatus.OK)
    }
}